File: src/sagacraft/core/priorities.py
"""Shared priority enums used across subsystems.

We intentionally keep one canonical priority scale to avoid drift between
plugins and event handling.
"""

from enum import IntEnum


class Priority(IntEnum):
    """Shared priority scale (lower values run first)."""

    CRITICAL = 0
    HIGH = 10
    NORMAL = 50
    LOW = 100

File: rust/sagacraft_rs/src/pyport/core/priorities.rs
#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
#[repr(i32)]
pub enum Priority {
    /// Lower values run first.
    Critical = 0,
    High = 10,
    Normal = 50,
    Low = 100,
}

impl Default for Priority {
    fn default() -> Self {
        Priority::Normal
    }
}
